src/clib/lib/config/conf.cpp

```diff
@@ -433,156 +371,135 @@ void conf_item_spec_set_default_value(conf_item_spec_type *conf_item_spec,
     return conf_instance->conf_class->class_name;
 }
 
-const char *
-conf_instance_get_item_value_ref(const conf_instance_type *conf_instance,
-                                 const char *item_name) {
-    if (!hash_has_key(conf_instance->items, item_name)) {
-        util_abort("%s: Instance %s of type %s has no item %s.\n", __func__,
-                   conf_instance->name, conf_instance->conf_class->class_name,
-                   item_name);
-    }
-    const conf_item_type *conf_item =
-        (const conf_item_type *)hash_get(conf_instance->items, item_name);
-    return conf_item->value;
+const char *conf_instance_get_item_value_ref(
+    std::shared_ptr<conf_instance_type> conf_instance, std::string item_name) {
+    return conf_instance->items[item_name]->value;
 }
-
 /** If the dt supports it, this function shall return the item value as an int.
     If the dt does not support it, the function will abort.
 */
-int conf_instance_get_item_value_int(const conf_instance_type *conf_instance,
-                                     const char *item_name) {
-    if (!hash_has_key(conf_instance->items, item_name))
+int conf_instance_get_item_value_int(
+    std::shared_ptr<conf_instance_type> conf_instance, std::string item_name) {
+    if (!conf_instance->items.count(item_name))
         util_abort("%s: Instance %s of type %s has no item %s.\n", __func__,
                    conf_instance->name, conf_instance->conf_class->class_name,
-                   item_name);
+                   item_name.c_str());
 
-    const conf_item_type *conf_item =
-        (const conf_item_type *)hash_get(conf_instance->items, item_name);
-    const conf_item_spec_type *conf_item_spec = conf_item->conf_item_spec;
-
-    return conf_data_get_int_from_string(conf_item_spec->dt, conf_item->value);
+    auto conf_item = conf_instance->items[item_name];
+    return conf_data_get_int_from_string(conf_item->conf_item_spec->dt,
+                                         conf_item->value);
 }
 
 /** If the dt supports it, this function shall return the item value as a double.
     If the dt does not support it, the function will abort.
 */
-double
-conf_instance_get_item_value_double(const conf_instance_type *conf_instance,
-                                    const char *item_name) {
-    if (!hash_has_key(conf_instance->items, item_name))
+double conf_instance_get_item_value_double(
+    std::shared_ptr<conf_instance_type> conf_instance, std::string item_name) {
+    if (!conf_instance->items.count(item_name))
         util_abort("%s: Instance %s of type %s has no item %s.\n", __func__,
                    conf_instance->name, conf_instance->conf_class->class_name,
-                   item_name);
-
-    const conf_item_type *conf_item =
-        (const conf_item_type *)hash_get(conf_instance->items, item_name);
-    const conf_item_spec_type *conf_item_spec =
-        (const conf_item_spec_type *)conf_item->conf_item_spec;
+                   item_name.c_str());
```

Comment:
This could be refactored into a function, where functions for int, double, ref and time_t will access the map through that.

Reply:
I agree, but I think we will rethink the whole design once we move to python.